### ISSUE TITLE
Interface being overridden  

### ISSUE THREAD

**Issue by Anishali2** [microsoft/TypeScript#52590]
# Bug Report

### 🔎 Search Terms

Interface override , interface not getting the actual types 

### 🕗 Version & Regression Information

- This changed between versions 4.0.5 and  4.9.4.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJIBUCeAHCBnZAbwChlkQ4BbCALmTzClAHMBuU5OAEy6nzzoMmINsQC+xBAHsQDZAEcArtEzIAvEQ6VQdACwAaLXAAedAAyGx7abLDIpAIwBWdDDnzrNZCtToByOBBgPAALP0Mybl5+fzMARgAmAGY4PC4YJ3COADpcgAolFWztEHUNRRAuCBhQCC5kAH4iMWQaQhKaQqhMYtAxAEpkAHoh5GA7UKlFABt6hBCIBABrZDB3AnF2IA)

### 💻 Code

```ts
interface ITypes {
  name: string;
  address: string;
}
const query = {
  min: 4,
  max: 0,
};
const obj: ITypes = {
  name: 'anish',
  address: '0123asdfj',
  ...(query.min == undefined ? {} :{min:query.min}) // it should check types 
};

```

### 🙁 Actual behavior

{
  "name": "anish",
  "address": "0123asdfj",
  "min": 4
} 

### 🙂 Expected behavior

{
  "name": "anish",
  "address": "0123asdfj",
  "min": 4 // it should throw error min is not defined in ITypes
}

**Comment by fatcerberus** [comment]
Almost positive it’s a known limitation that excess property checks (which FWIW you shouldn’t rely on as a type check, it only applies to directly-assigned object literals - see #12936) don’t work with spread.
